### PR TITLE
IT Wazuh DB: Add coverage to new commands

### PR DESCRIPTION
**Pull request by damarisg**
## Description

New changes in Wazuh DB add new commands to cover:
Attached is the list of PRs who worked on each of the commands, there you can find in detail each of the implemented cases. 

| Command | PR Detailed |
|--|--|
| sync-agent-groups-get| https://github.com/wazuh/wazuh-qa/pull/2626
| set-agent-groups| https://github.com/wazuh/wazuh-qa/pull/2602
| get-groups-integrity| https://github.com/wazuh/wazuh-qa/pull/2607
| global backup| https://github.com/wazuh/wazuh-qa/pull/2637 
| wazuhdb getconfig| https://github.com/wazuh/wazuh-qa/pull/2627

<table>
  <tr>
    <th>Related issues</th>
    <th><a href="[url](https://github.com/wazuh/wazuh/issues/10771)">#10771 (wazuh)</a></th>
  </tr>
</table>

## Configuration options
|OS | Version | Branch | Type |
|--|--|--|--|
| Centos | 4.4.0 | dev-10771-agent-groups-files-to-wazuh-db | Manager|

**Local Internal Options:** No Local Internal Options used

---
## Required
- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.
- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.
- [x] The test is documented in wazuh-qa/docs.

### CODE REVIEW

**Comment by Deblintrake09** [comment]
## Tests Results
| Test Path | Os/Type | Execution Type | Results | Date | By |Notes|
|--|--|--|--|--|--|--|
| test/integration/test_wazuh_db/   | CentOS - Manager | Local | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8265870/R1.zip)[:green_circle:](https://github.com/wazuh/wazuh-qa/files/8265873/R3.zip) [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8265872/R2.zip)| 16/03/2022 |  @Deblintrake09 | Run all folder except `test_wazuh_db.py`  |
| test/integration/test_wazuh_db/   | CentOS - Manager | Jenkins | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/23725/)[:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/23726/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/23727/)| 16/03/2022 |  @Deblintrake09 | Run all folder except `test_wazuh_db.py`

File: deps/wazuh_testing/wazuh_testing/tools/file.py [hunk 1]
```diff
@@ -51,6 +51,22 @@ def read_yaml(file_path):
...
+        filter (str): filder to extract some part of yaml
+        list: a list with previous information
+    
+    Returns:
+       list: Yaml structure.
+    """
+    list.append((read_yaml(file_path), file_path + filter))
+    return list
+
+
```
Comment (damarisg) [inline deps/wazuh_testing/wazuh_testing/tools/file.py:69]: If we don't use this `method `but this block is repeat on each test, 

> with open(messages_file) as f:
>     module_tests.append((yaml.safe_load(f), messages_file.split('_')[0]))
> 
Could we add this content on a this method an use it on each test?

File: tests/integration/test_wazuh_db/data/global/wazuhdb_getconfig.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  test_case:
+  -
+    input: 'wazuhdb getconfig wrong_parameter'
+    output: 'err Failed reading wazuh-db config'
+-
+  name: 'Get internal config'
+  test_case:
+  -
+    input: 'wazuhdb getconfig internal'
+    output: "{'wazuh_db': {'commit_time_max': 60, 'commit_time_min': 10, 'open_db_limit': 64, 'worker_pool_size': 8}}"
```
Comment (damarisg) [inline tests/integration/test_wazuh_db/data/global/wazuhdb_getconfig.yaml:25]: Is there any reason to disappear the `'sock_queue_size': 128` value?

File: tests/integration/test_wazuh_db/test_get_groups_integrity.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        hash = case_data["invalid_hash"]
+    else:
+        response = query_wdb(f'global sync-agent-groups-get {{"last_id": 0, "condition": "all", "get_global_hash": true, \
+                             "set_synced": false, "agent_delta_registration": 0}}')
+        response = response[0]
+        hash = response["hash"]
+        if "no_hash" in case_data:
+            response = str(response)
+            assert output in response , f'Unexpected response: got {response}, but expected {output}'
+            return
```
Comment (damarisg) [inline tests/integration/test_wazuh_db/test_get_groups_integrity.py:148]: Check with Core Team if it is correct that now we receive a hash value.

File: tests/integration/test_wazuh_db/data/global/wazuhdb_getconfig.yaml (hunk 2 above)
Comment (Deblintrake09) [reply tests/integration/test_wazuh_db/data/global/wazuhdb_getconfig.yaml:25]: It was removed by Core team, change is expected.

File: tests/integration/test_wazuh_db/test_get_groups_integrity.py (hunk 3 above)
Comment (Deblintrake09) [reply tests/integration/test_wazuh_db/test_get_groups_integrity.py:147]: Change was made by Core team, so there's always a hash value

**Review by Deblintrake09 — COMMENTED** [review]
applied requested change. refactored function that opens yaml file.
Approved

**Review by Deblintrake09 — APPROVED** [review]

File: tests/integration/test_wazuh_db/test_get_groups_integrity.py (hunk 3 above)
Comment (damarisg) [reply tests/integration/test_wazuh_db/test_get_groups_integrity.py:147]: It's correct!

File: tests/integration/test_wazuh_db/data/global/wazuhdb_getconfig.yaml (hunk 2 above)
Comment (damarisg) [reply tests/integration/test_wazuh_db/data/global/wazuhdb_getconfig.yaml:25]: Checked with Core team.

**Comment by damarisg** [comment]
## Tests Results
| Test Path | Os/Type | Execution Type | Results | Date | By |Notes|
|--|--|--|--|--|--|--|
| test/integration/test_wazuh_db/   | CentOS - Manager | Jenkins | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/23824/)[:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/23825/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/23826/)| 17/03/2022 | Seyla| Run all folder except `test_wazuh_db.py`. There are still some tests to be fixed in another branch and they will fail here.

**Pull request merged**